# Indian Start-Up Funding Ecosystem

## Business Understanding
Objective: The primary objective is to venture into the Indian start-up ecosystem. This involves understanding the landscape of start-ups in India, their funding patterns, and the key players in the ecosystem (both start-ups and investors).

Problem Definition: The problem can be defined as analyzing the funding received by Indian start-ups from 2018 to 2021. This involves understanding the trends in funding, identifying the sectors or industries that are attracting the most investment, and recognizing the major investors in the ecosystem.

Data Understanding: The data that will be used for this analysis includes details of the start-ups, the funding amounts they received, and information about the investors. This data will be provided separately for each year from 2018 to 2021 via three different sources.

Plan: The preliminary plan would involve cleaning and preprocessing the data, conducting exploratory data analysis to understand trends and patterns, and possibly building predictive models to forecast future trends in the start-up ecosystem.

Success Criteria: The success of this project could be measured by the team’s ability to gain insights that help in making informed decisions about venturing into the Indian start-up ecosystem. This could involve identifying promising sectors, understanding the competitive landscape, and recognizing potential investment opportunities.

## Hypothesis Test
Null Hypothesis (H0): The location of a start-up in India does not affect the amount of funding it receives.
Alternative Hypothesis (H1): The location of a start-up in India does affect the amount of funding it receives.

Null Hypothesis (H0): There is no difference in the amount of funding received by start-ups across different sectors.
Alternative Hypothesis (H1): There is a difference in the amount of funding received by start-ups across different sectors.

Null Hypothesis (H0): The size of a start-up (in terms of employees or customers) does not affect the amount of funding it receives.
Alternative Hypothesis (H1): The size of a start-up (in terms of employees or customers) does affect the amount of funding it receives.

Null Hypothesis (H0): All investors contribute equally to the funding of start-ups.
Alternative Hypothesis (H1): Some investors contribute more to the funding of start-ups than others.

Null Hypothesis (H0): The amount of funding received by Indian start-ups has not changed from 2018 to 2021.
Alternative Hypothesis (H1): The amount of funding received by Indian start-ups has increased or decreased from 2018 to 2021.


## Relevant Questions
1 - How has the funding trend for Indian start-ups changed from 2018 to 2021? Are there any noticeable patterns or trends?

2 - Which sectors or industries received the most funding? Are there any sectors that are emerging as new favorites for investors?

3 - Who are the major investors in the Indian start-up ecosystem? Are there any investors who are particularly active or influential?

4 - Are there any specific cities or regions in India that are attracting more start-ups or funding?

5 - Is there a correlation between the size of the start-up (in terms of employees or customers) and the amount of funding received?

6 - Which start-ups have shown the most growth in terms of funding received over the years?

7 - Is there a correlation between the amount of funding received and the success of the start-up?

8 - Can we identify common investment strategies among the major investors?

In [7]:
# Import requisite libraries
import pyodbc    
from dotenv import dotenv_values
import warnings 
warnings.filterwarnings('ignore')
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Data Importation

### Importing and Cleaning 2018 Dataset

In [28]:
# Access the 2018 dataset
url = 'https://raw.githubusercontent.com/Azubi-Africa/Career_Accelerator_LP1-Data_Analysis/main/startup_funding2018.csv'

dat1 = pd.read_csv(url)
dat1

,Company Name,Industry,Round/Series,Amount,Location,About Company
0,TheCollegeFever,"Brand Marketing, Event Promotion, Marketing, S...",Seed,250000,"Bangalore, Karnataka, India","TheCollegeFever is a hub for fun, fiesta and f..."
1,Happy Cow Dairy,"Agriculture, Farming",Seed,"₹40,000,000","Mumbai, Maharashtra, India",A startup which aggregates milk from dairy far...
2,MyLoanCare,"Credit, Financial Services, Lending, Marketplace",Series A,"₹65,000,000","Gurgaon, Haryana, India",Leading Online Loans Marketplace in India
3,PayMe India,"Financial Services, FinTech",Angel,2000000,"Noida, Uttar Pradesh, India",PayMe India is an innovative FinTech organizat...
4,Eunimart,"E-Commerce Platforms, Retail, SaaS",Seed,—,"Hyderabad, Andhra Pradesh, India",Eunimart is a one stop solution for merchants ...
...,...,...,...,...,...,...
521,Udaan,"B2B, Business Development, Internet, Marketplace",Series C,225000000,"Bangalore, Karnataka, India","Udaan is a B2B trade platform, designed specif..."
522,Happyeasygo Group,"Tourism, Travel",Series A,—,"Haryana, Haryana, India",HappyEasyGo is an online travel domain.
523,Mombay,"Food and Beverage, Food Delivery, Internet",Seed,7500,"Mumbai, Maharashtra, India",Mombay is a unique opportunity for housewives ...
524,Droni Tech,Information Technology,Seed,"₹35,000,000","Mumbai, Maharashtra, India",Droni Tech manufacture UAVs and develop softwa...


In [29]:
# adding years
year_1 = 2018
dat1['Year'] = year_1


In [48]:
dat1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
 6   Year           526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


In [49]:
dat1["Amount"].unique()

array(['250000', 584000.0, 949000.0, '2000000', '—', '1600000', 233600.0,
       730000.0, 1460000.0, '150000', '1100000', 7300.0, '6000000',
       '650000', 511000.0, 934400.0, 292000.0, '1000000', '5000000',
       '4000000', 438000.0, '2800000', '1700000', '1300000', 73000.0,
       182500.0, 219000.0, '500000', 1518400.0, 657000.0, '13400000',
       365000.0, 385440.0, 116800.0, 876.0, '9000000', '100000', '20000',
       '120000', 496400.0, 4993200.0, '$143145', 8760000.0, '$742000000',
       14600000.0, 29200000.0, '$3980000', '$10000', 1460.0, 3650000.0,
       '$1000000000', '$7000000', '$35000000', 8030000.0, '$28500000',
       '$2000000', 3504000.0, 1752000.0, '$2400000', '$30000000',
       36500000.0, '$23000000', '$150000', '$11000000', 642400.0,
       '$3240000', 876000.0, '$540000000', 9490000.0, 23360000.0,
       '$900000', '$10000000', '$1500000', 1022000.0, '$1000000',
       '$5000000', '$14000000', 1496500.0, '$100000000', 17520.0,
       75920000.0, '$800000'

In [52]:
dat1[dat1["Amount"].str.startswith("$")]

ValueError: Cannot mask with non-boolean array containing NA / NaN values

In [ ]:

dat1['Amount'] = dat1['Amount'].str.replace(',', '')
dat1

,Company Name,Industry,Round/Series,Amount,Location,About Company,Year
0,TheCollegeFever,"Brand Marketing, Event Promotion, Marketing, S...",Seed,250000,"Bangalore, Karnataka, India","TheCollegeFever is a hub for fun, fiesta and f...",2018
1,Happy Cow Dairy,"Agriculture, Farming",Seed,₹40000000,"Mumbai, Maharashtra, India",A startup which aggregates milk from dairy far...,2018
2,MyLoanCare,"Credit, Financial Services, Lending, Marketplace",Series A,₹65000000,"Gurgaon, Haryana, India",Leading Online Loans Marketplace in India,2018
3,PayMe India,"Financial Services, FinTech",Angel,2000000,"Noida, Uttar Pradesh, India",PayMe India is an innovative FinTech organizat...,2018
4,Eunimart,"E-Commerce Platforms, Retail, SaaS",Seed,—,"Hyderabad, Andhra Pradesh, India",Eunimart is a one stop solution for merchants ...,2018
...,...,...,...,...,...,...,...
521,Udaan,"B2B, Business Development, Internet, Marketplace",Series C,225000000,"Bangalore, Karnataka, India","Udaan is a B2B trade platform, designed specif...",2018
522,Happyeasygo Group,"Tourism, Travel",Series A,—,"Haryana, Haryana, India",HappyEasyGo is an online travel domain.,2018
523,Mombay,"Food and Beverage, Food Delivery, Internet",Seed,7500,"Mumbai, Maharashtra, India",Mombay is a unique opportunity for housewives ...,2018
524,Droni Tech,Information Technology,Seed,₹35000000,"Mumbai, Maharashtra, India",Droni Tech manufacture UAVs and develop softwa...,2018


In [ ]:
dat1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
 6   Year           526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


In [ ]:
# Define a function to convert rupees to dollars
def convert_to_dollars(Amount):
    if Amount.startswith('₹'):
        return float(Amount[1:]) * 0.0146
    else:
        return Amount
 
# Apply the conversion function to the 'amount' column
dat1['Amount'] = dat1['Amount'].apply(convert_to_dollars)
dat1

,Company Name,Industry,Round/Series,Amount,Location,About Company,Year
0,TheCollegeFever,"Brand Marketing, Event Promotion, Marketing, S...",Seed,250000,"Bangalore, Karnataka, India","TheCollegeFever is a hub for fun, fiesta and f...",2018
1,Happy Cow Dairy,"Agriculture, Farming",Seed,584000.0,"Mumbai, Maharashtra, India",A startup which aggregates milk from dairy far...,2018
2,MyLoanCare,"Credit, Financial Services, Lending, Marketplace",Series A,949000.0,"Gurgaon, Haryana, India",Leading Online Loans Marketplace in India,2018
3,PayMe India,"Financial Services, FinTech",Angel,2000000,"Noida, Uttar Pradesh, India",PayMe India is an innovative FinTech organizat...,2018
4,Eunimart,"E-Commerce Platforms, Retail, SaaS",Seed,—,"Hyderabad, Andhra Pradesh, India",Eunimart is a one stop solution for merchants ...,2018
...,...,...,...,...,...,...,...
521,Udaan,"B2B, Business Development, Internet, Marketplace",Series C,225000000,"Bangalore, Karnataka, India","Udaan is a B2B trade platform, designed specif...",2018
522,Happyeasygo Group,"Tourism, Travel",Series A,—,"Haryana, Haryana, India",HappyEasyGo is an online travel domain.,2018
523,Mombay,"Food and Beverage, Food Delivery, Internet",Seed,7500,"Mumbai, Maharashtra, India",Mombay is a unique opportunity for housewives ...,2018
524,Droni Tech,Information Technology,Seed,511000.0,"Mumbai, Maharashtra, India",Droni Tech manufacture UAVs and develop softwa...,2018


In [ ]:
dat1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
 6   Year           526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


### Loading and Cleaning 2019 datasets

In [ ]:
# Access the 2019 dataset

dat2 = pd.read_csv('dataset/startup_funding2019.csv')
dat2

,Company/Brand,Founded,HeadQuarter,Sector,What it does,Founders,Investor,Amount($),Stage
0,Bombay Shaving,NaN,NaN,Ecommerce,Provides a range of male grooming products,Shantanu Deshpande,Sixth Sense Ventures,"$6,300,000",NaN
1,Ruangguru,2014.0,Mumbai,Edtech,A learning platform that provides topic-based ...,"Adamas Belva Syah Devara, Iman Usman.",General Atlantic,"$150,000,000",Series C
2,Eduisfun,NaN,Mumbai,Edtech,It aims to make learning fun via games.,Jatin Solanki,"Deepak Parekh, Amitabh Bachchan, Piyush Pandey","$28,000,000",Fresh funding
3,HomeLane,2014.0,Chennai,Interior design,Provides interior designing solutions,"Srikanth Iyer, Rama Harinath","Evolvence India Fund (EIF), Pidilite Group, FJ...","$30,000,000",Series D
4,Nu Genes,2004.0,Telangana,AgriTech,"It is a seed company engaged in production, pr...",Narayana Reddy Punyala,Innovation in Food and Agriculture (IFA),"$6,000,000",NaN
...,...,...,...,...,...,...,...,...,...
84,Infra.Market,NaN,Mumbai,Infratech,It connects client requirements to their suppl...,"Aaditya Sharda, Souvik Sengupta","Tiger Global, Nexus Venture Partners, Accel Pa...","$20,000,000",Series A
85,Oyo,2013.0,Gurugram,Hospitality,Provides rooms for comfortable stay,Ritesh Agarwal,"MyPreferred Transformation, Avendus Finance, S...","$693,000,000",NaN
86,GoMechanic,2016.0,Delhi,Automobile & Technology,Find automobile repair and maintenance service...,"Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...",Sequoia Capital,"$5,000,000",Series B
87,Spinny,2015.0,Delhi,Automobile,Online car retailer,"Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...","Norwest Venture Partners, General Catalyst, Fu...","$50,000,000",NaN


In [ ]:
dat2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount($)      89 non-null     object 
 8   Stage          43 non-null     object 
dtypes: float64(1), object(8)
memory usage: 6.4+ KB


In [ ]:
# adding years
year_2 = 2019
dat2['Year'] = year_2


In [ ]:
#Renaming amount column in dataset 2#

dat2 = dat2.rename(columns={'Amount($)':'Amount'})
dat2

,Company/Brand,Founded,HeadQuarter,Sector,What it does,Founders,Investor,Amount,Stage,Year
0,Bombay Shaving,NaN,NaN,Ecommerce,Provides a range of male grooming products,Shantanu Deshpande,Sixth Sense Ventures,"$6,300,000",NaN,2019
1,Ruangguru,2014.0,Mumbai,Edtech,A learning platform that provides topic-based ...,"Adamas Belva Syah Devara, Iman Usman.",General Atlantic,"$150,000,000",Series C,2019
2,Eduisfun,NaN,Mumbai,Edtech,It aims to make learning fun via games.,Jatin Solanki,"Deepak Parekh, Amitabh Bachchan, Piyush Pandey","$28,000,000",Fresh funding,2019
3,HomeLane,2014.0,Chennai,Interior design,Provides interior designing solutions,"Srikanth Iyer, Rama Harinath","Evolvence India Fund (EIF), Pidilite Group, FJ...","$30,000,000",Series D,2019
4,Nu Genes,2004.0,Telangana,AgriTech,"It is a seed company engaged in production, pr...",Narayana Reddy Punyala,Innovation in Food and Agriculture (IFA),"$6,000,000",NaN,2019
...,...,...,...,...,...,...,...,...,...,...
84,Infra.Market,NaN,Mumbai,Infratech,It connects client requirements to their suppl...,"Aaditya Sharda, Souvik Sengupta","Tiger Global, Nexus Venture Partners, Accel Pa...","$20,000,000",Series A,2019
85,Oyo,2013.0,Gurugram,Hospitality,Provides rooms for comfortable stay,Ritesh Agarwal,"MyPreferred Transformation, Avendus Finance, S...","$693,000,000",NaN,2019
86,GoMechanic,2016.0,Delhi,Automobile & Technology,Find automobile repair and maintenance service...,"Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...",Sequoia Capital,"$5,000,000",Series B,2019
87,Spinny,2015.0,Delhi,Automobile,Online car retailer,"Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...","Norwest Venture Partners, General Catalyst, Fu...","$50,000,000",NaN,2019


In [ ]:
# Define a function to convert rupees to dollars
def convert_to_dollars(Amount):
    if Amount.startswith('₹'):
        return float(Amount[1:]) * 0.0142
    else:
        return Amount
 
# Apply the conversion function to the 'amount' column
dat2['Amount'] = dat2['Amount'].apply(convert_to_dollars)
dat2

,Company/Brand,Founded,HeadQuarter,Sector,What it does,Founders,Investor,Amount,Stage,Year
0,Bombay Shaving,NaN,NaN,Ecommerce,Provides a range of male grooming products,Shantanu Deshpande,Sixth Sense Ventures,"$6,300,000",NaN,2019
1,Ruangguru,2014.0,Mumbai,Edtech,A learning platform that provides topic-based ...,"Adamas Belva Syah Devara, Iman Usman.",General Atlantic,"$150,000,000",Series C,2019
2,Eduisfun,NaN,Mumbai,Edtech,It aims to make learning fun via games.,Jatin Solanki,"Deepak Parekh, Amitabh Bachchan, Piyush Pandey","$28,000,000",Fresh funding,2019
3,HomeLane,2014.0,Chennai,Interior design,Provides interior designing solutions,"Srikanth Iyer, Rama Harinath","Evolvence India Fund (EIF), Pidilite Group, FJ...","$30,000,000",Series D,2019
4,Nu Genes,2004.0,Telangana,AgriTech,"It is a seed company engaged in production, pr...",Narayana Reddy Punyala,Innovation in Food and Agriculture (IFA),"$6,000,000",NaN,2019
...,...,...,...,...,...,...,...,...,...,...
84,Infra.Market,NaN,Mumbai,Infratech,It connects client requirements to their suppl...,"Aaditya Sharda, Souvik Sengupta","Tiger Global, Nexus Venture Partners, Accel Pa...","$20,000,000",Series A,2019
85,Oyo,2013.0,Gurugram,Hospitality,Provides rooms for comfortable stay,Ritesh Agarwal,"MyPreferred Transformation, Avendus Finance, S...","$693,000,000",NaN,2019
86,GoMechanic,2016.0,Delhi,Automobile & Technology,Find automobile repair and maintenance service...,"Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...",Sequoia Capital,"$5,000,000",Series B,2019
87,Spinny,2015.0,Delhi,Automobile,Online car retailer,"Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...","Norwest Venture Partners, General Catalyst, Fu...","$50,000,000",NaN,2019


In [ ]:
dat2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount         89 non-null     object 
 8   Stage          43 non-null     object 
 9   Year           89 non-null     int64  
dtypes: float64(1), int64(1), object(8)
memory usage: 7.1+ KB


In [ ]:
#removing dollar sign
dat2['Amount']= dat2['Amount'].str.replace('$', '')
dat2

,Company/Brand,Founded,HeadQuarter,Sector,What it does,Founders,Investor,Amount,Stage,Year
0,Bombay Shaving,NaN,NaN,Ecommerce,Provides a range of male grooming products,Shantanu Deshpande,Sixth Sense Ventures,"6,300,000",NaN,2019
1,Ruangguru,2014.0,Mumbai,Edtech,A learning platform that provides topic-based ...,"Adamas Belva Syah Devara, Iman Usman.",General Atlantic,"150,000,000",Series C,2019
2,Eduisfun,NaN,Mumbai,Edtech,It aims to make learning fun via games.,Jatin Solanki,"Deepak Parekh, Amitabh Bachchan, Piyush Pandey","28,000,000",Fresh funding,2019
3,HomeLane,2014.0,Chennai,Interior design,Provides interior designing solutions,"Srikanth Iyer, Rama Harinath","Evolvence India Fund (EIF), Pidilite Group, FJ...","30,000,000",Series D,2019
4,Nu Genes,2004.0,Telangana,AgriTech,"It is a seed company engaged in production, pr...",Narayana Reddy Punyala,Innovation in Food and Agriculture (IFA),"6,000,000",NaN,2019
...,...,...,...,...,...,...,...,...,...,...
84,Infra.Market,NaN,Mumbai,Infratech,It connects client requirements to their suppl...,"Aaditya Sharda, Souvik Sengupta","Tiger Global, Nexus Venture Partners, Accel Pa...","20,000,000",Series A,2019
85,Oyo,2013.0,Gurugram,Hospitality,Provides rooms for comfortable stay,Ritesh Agarwal,"MyPreferred Transformation, Avendus Finance, S...","693,000,000",NaN,2019
86,GoMechanic,2016.0,Delhi,Automobile & Technology,Find automobile repair and maintenance service...,"Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...",Sequoia Capital,"5,000,000",Series B,2019
87,Spinny,2015.0,Delhi,Automobile,Online car retailer,"Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...","Norwest Venture Partners, General Catalyst, Fu...","50,000,000",NaN,2019


In [ ]:
# remove commas
dat2['Amount']= dat2['Amount'].str.replace(',', '', regex= True)

dat2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount         89 non-null     object 
 8   Stage          43 non-null     object 
 9   Year           89 non-null     int64  
dtypes: float64(1), int64(1), object(8)
memory usage: 7.1+ KB


In [ ]:
dat2['Amount'].unique()

array(['6300000', '150000000', '28000000', '30000000', '6000000',
       'Undisclosed', '1000000', '20000000', '275000000', '22000000',
       '5000000', '140500', '540000000', '15000000', '182700', '12000000',
       '11000000', '15500000', '1500000', '5500000', '2500000', '140000',
       '230000000', '49400000', '32000000', '26000000', '150000',
       '400000', '2000000', '100000000', '8000000', '100000', '50000000',
       '120000000', '4000000', '6800000', '36000000', '5700000',
       '25000000', '600000', '70000000', '60000000', '220000', '2800000',
       '2100000', '7000000', '311000000', '4800000', '693000000',
       '33000000'], dtype=object)

In [ ]:
dat2_undisclosed = dat2[dat2['Amount']=='Undisclosed']
dat2_undisclosed

,Company/Brand,Founded,HeadQuarter,Sector,What it does,Founders,Investor,Amount,Stage,Year
5,FlytBase,NaN,Pune,Technology,A drone automation platform,Nitin Gupta,Undisclosed,Undisclosed,NaN,2019
6,Finly,NaN,Bangalore,SaaS,It builds software products that makes work si...,"Vivek AG, Veekshith C Rai","Social Capital, AngelList India, Gemba Capital...",Undisclosed,NaN,2019
10,Cub McPaws,2010.0,Mumbai,E-commerce & AR,A B2C brand that focusses on premium and comf...,"Abhay Bhat, Kinnar Shah",Venture Catalysts,Undisclosed,NaN,2019
14,Open Secret,NaN,NaN,Food tech,It produces and sells top quality snacks,"Ahana Gautam, Udit Kejriwal",Matrix Partners,Undisclosed,NaN,2019
19,Azah Personal Care Pvt. Ltd.,2018.0,Gurugram,Health,Aims to solve some problems in the feminine hy...,"Mohammed, Shashwat Diesh","Kunal Bahl, Rohit Bansal.",Undisclosed,Pre series A,2019
23,DROR Labs Pvt. Ltd,2018.0,Delhi,Safety tech,It uses technology to create a trust-based net...,"Dhiraj Naubhar, Dheeraj Bansal",Inflection Point Ventures,Undisclosed,NaN,2019
32,Pumpkart,2014.0,Chandigarh,E-marketplace,B2B model for appliances and electrical products,KS Bhatia,Dinesh Dua,Undisclosed,NaN,2019
45,Afinoz,NaN,Noida,Fintech,Online financial marketplace for customized ra...,Rachna Suneja,Fintech innovation lab,Undisclosed,NaN,2019
54,Ninjacart,2015.0,NaN,B2B Supply Chain,It connects producers of food directly to reta...,"Thirukumaran Nagarajanin, Vasudevan Chinnathambi","Walmart, Flipkart",Undisclosed,NaN,2019
55,Binca Games,2014.0,Mumbai,Games,It offers games that are available across Indi...,"Rubianca Wadhwa, Sahil Wadhwa",Sunil Gavaskar,Undisclosed,NaN,2019


In [ ]:
# save undisclosed data in a csv
dat2_undisclosed.to_csv('../analysis-into-indian-startup-ecosystem/dataset/undisclosed_amount_data.csv', index=False)

OSError: Cannot save file into a non-existent directory: '../analysis-into-indian-startup-ecosystem/dataset'

In [ ]:
dat2.info()

: 

In [ ]:
dat2 = dat2[dat2['Amount']!='Undisclosed']
dat2['Amount'].unique()

: 

In [ ]:
dat2.info()

: 

In [ ]:
# convert amount to float
dat2['Amount']= dat2['Amount'].astype(float)
dat2

: 

### Importing and cleaning 2020 and 2021 datasets

In [ ]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')

# Get the values for the credentials you set in the '.env' file
server = environment_variables.get("SERVER_NAME")
database = environment_variables.get("DATABASE_NAME")
username = environment_variables.get("DB_USERNAME")
password = environment_variables.get("PASSWORD")

: 

In [ ]:

connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password};MARS_Connection=yes;MinProtocolVersion=TLSv1.2;"


: 

In [ ]:
connection = pyodbc.connect(connection_string)

: 

In [ ]:
# select data from 2020
query = "SELECT * FROM LP1_startup_funding2020"

dat3 = pd.read_sql(query, connection)
dat3.head()

: 

In [ ]:
dat3.info()

: 

In [ ]:
# adding years
year_3 = 2020
dat3['Year'] = year_3
dat3

: 

In [ ]:
# loading data from 2021
query = "SELECT * FROM LP1_startup_funding2021"

dat4 = pd.read_sql(query, connection)
dat4.head()

: 

In [ ]:
# adding years
year_4 = 2021
dat4['Year'] = year_4
dat4


: 

In [ ]:
dat4.info()

: 

In [ ]:
dat3=dat3.drop('column10', axis=1)
dat3

: 

## Merging datasets

In [ ]:
#merging 2020 and 2021 datasets
merged_table= pd.concat([dat3, dat4], ignore_index=True)
merged_table

: 

In [ ]:
#renaming company name columns
dat1.rename(columns={'Company Name':'Company_Brand'}, inplace= True)

dat2.rename(columns={'Company/Brand':'Company_Brand'}, inplace= True)

: 

In [ ]:
#renaming round/series to name columns
dat1.rename(columns={'Round/Series':'Stage'}, inplace= True)

: 

In [ ]:
#renaming Industry to name Sector
dat1.rename(columns={'Industry':'Sector'}, inplace= True)

: 

In [ ]:
#renaming Industry to name Sector
dat1.rename(columns={'About Company':'What it does'}, inplace= True)

: 

In [ ]:
#renaming Industry to name Sector
dat1.rename(columns={'Location':'HeadQuarter'}, inplace= True)

: 

In [ ]:
# extracting the locations in dat1
dat1['HeadQuarter']= dat1['HeadQuarter'].str.split(',').str[0]

: 

In [ ]:
#printing columns to compare if the column names are matching
print(dat1.columns)
print(dat2.columns)
print(merged_table.columns)


: 

In [ ]:
dat1.head(3)

: 

In [ ]:
dat2['HeadQuarter'].unique()

NameError: name 'dat2' is not defined

In [ ]:
dat2.head(3)

: 

In [ ]:
merged_table.head(3)

: 

In [ ]:
merged_table.info()

: 

In [ ]:
#renaming What_it_does to What it does
merged_table.rename(columns={'What_it_does':'What it does'}, inplace= True)
merged_table

: 

In [ ]:
#merging 2019,2020,2021
merger= pd.concat([dat2, merged_table], ignore_index=True)
merger

: 

In [ ]:
merger.info()

: 

In [ ]:
#merging 2018,2019,2020,2021
data_df= pd.concat([dat1, merger], ignore_index=True)
data_df

: 

In [ ]:
data_df.info()

: 

### Data Cleaning by Columns

### Cleaning Amount Column

In [ ]:
data_df['Amount'].unique()

: 

In [ ]:
# removing the $
data_df['Amount']= data_df['Amount'].apply(lambda x:str(x).replace('$',''))

# .apply(lambda x:str(x).replace('$','')
data_df.info()




: 

In [ ]:
data_df['Amount'].unique()

: 

In [ ]:
data_df['Amount'].unique()

: 

In [ ]:
# removing the $
data_df['Amount']= data_df['Amount'].apply(lambda x:str(x).replace(',',''))

# .apply(lambda x:str(x).replace(',','')
data_df.info()


: 

In [ ]:
# Filter the records where the amount column contains 'undisclosed'
undisclosed_data = data_df[data_df['Amount'] == 'Undisclosed']
undisclosed_data


: 

In [ ]:
undisclosed_data.info()

: 

In [ ]:
# Remove the undisclosed records from Data_df
data_df = data_df[data_df['Amount']!='Undisclosed']
data_df['Amount'].unique()

: 

In [ ]:
# checking data that has amount has upsparks
data_df[data_df['Amount'] == 'Upsparks']


: 

###### the script above printed two rows which are duplicates. so one of them were going to actually be removed
###### then the data in the following columns ; Sector, stage, Amount and Headquarters do not give enough justification to keep


In [ ]:
data_df = data_df[data_df['Amount'] != 'Upsparks']

: 

In [ ]:
# Remove dashes from the 'Amount' column
data_df['Amount'] = data_df['Amount'].replace('—', '')
data_df['Amount'].unique()

: 

In [ ]:
# Remove space from the 'Amount' column and convert to 0
data_df['Amount'] = data_df['Amount'].replace('', 0)
data_df['Amount'].unique()

: 

In [ ]:
#removing Series C from Amount column
data_df[data_df['Amount'] == 'Series C']

: 

In [ ]:
# Remove rows where the 'Amount' column is equal to 'Series C'
# These rows have been mislabeled and are duplicated.
data_df = data_df[data_df['Amount'] != 'Series C']


: 

In [ ]:
data_df[data_df['Amount'] == 'Seed']

: 

In [ ]:
# Remove rows where the 'Amount' column is equal to 'Seed'
# These rows have been mislabeled 
data_df = data_df[data_df['Amount'] != 'Seed']


: 

In [ ]:
# Convert undisclosed_data to DataFrame if it's not already
if not isinstance(undisclosed_data, pd.DataFrame):
    undisclosed_data = pd.DataFrame(undisclosed_data)

# Filter the additional records where the amount column contains 'Undisclosed' (both capital and lowercase)
additional_undisclosed_data = data_df[data_df['Amount'].str.lower() == 'undisclosed']

# Append the additional undisclosed records to the existing undisclosed_data
undisclosed_data = pd.concat([undisclosed_data, additional_undisclosed_data], ignore_index=True)


: 

In [ ]:
#remove undisclosed from amount
data_df=  data_df[data_df['Amount'].str.lower() != 'undisclosed']
data_df.info()


: 

In [ ]:
data_df[data_df['Amount'] == 'ah! Ventures']


: 

In [ ]:
#remove ah! Ventures from amount
data_df=  data_df[data_df['Amount'] != 'ah! Ventures']
data_df.info()


: 

In [ ]:
data_df[data_df['Amount'] == 'Pre-series A']

: 

In [ ]:
#remove ah! Ventures from amount
data_df=  data_df[data_df['Amount'] != 'Pre-series A']
data_df.info()

: 

In [ ]:
data_df[data_df['Amount'] == 'ITO Angel Network LetsVenture']

: 

In [ ]:
#remove ITO Angel Network LetsVenture from amount
data_df=  data_df[data_df['Amount'] != 'ITO Angel Network LetsVenture']
data_df.info()

: 

In [ ]:
data_df[data_df['Amount'] == 'JITO Angel Network LetsVenture']

: 

In [ ]:
#remove ITO Angel Network LetsVenture from amount
data_df=  data_df[data_df['Amount'] != 'JITO Angel Network LetsVenture']
data_df.info()

: 

In [ ]:
data_df['Amount'].unique()

: 

In [ ]:
data_df['Amount'].isna().sum()

: 

In [ ]:
data_df[data_df['Amount'] == 'None']

: 

In [ ]:
data_df = data_df[data_df['Amount'] != 'None']
data_df[data_df['Amount'] == 'None']

: 

In [ ]:
data_df.info()

: 

In [ ]:
# Remove non-numeric characters and the dollar sign ('$')
data_df['Amount'] = data_df['Amount'].apply(lambda x: str(x).replace('$', '') if pd.notna(x) else x)


: 

In [ ]:
# Convert 'Amount' column to float 
data_df['Amount'] = data_df['Amount'].astype(float)
data_df.info()

: 

### Cleaning the Company_Brand column


In [ ]:
# Get unique values in the Company_Brand column and sort them
sorted(data_df['Company_Brand'].unique())



: 

In [ ]:
# Get unique values in the Company_Brand column
unique_company_brands = data_df['Company_Brand'].unique()

# Convert the unique values array to a list
unique_company_brands_list = unique_company_brands.tolist()

# Print the list of unique company brands
unique_company_brands_list


: 

### Cleaning the Sector column

In [ ]:
data_df['Sector'].unique()

: 

In [ ]:
# Get unique values in the Company_Brand column
unique_sectors = data_df['Sector'].unique()

# Convert the unique values array to a list
unique_sectors_list = unique_sectors.tolist()

# Print the list of unique company brands
unique_sectors_list


: 

In [ ]:
# Filter the data where the "Sector" column is equal to "Manchester, Greater Manchester"
data_df[data_df['Sector'] == "Manchester, Greater Manchester"]

: 

In [ ]:
# Swap values between 'Sector' and 'HeadQuarter' columns for the row where 'Company_Brand' is 'Peak'
data_df.loc[data_df['Company_Brand'] == 'Peak', ['Sector', 'HeadQuarter']] = data_df.loc[data_df['Company_Brand'] == 'Peak', ['HeadQuarter', 'Sector']].values

# Filter the data where the "Company_Brand" column is equal to Peak
data_df[data_df['Company_Brand'] == "Peak"]

: 

In [ ]:
# Get unique values in the Company_Brand column
unique_sectors = data_df['Sector'].unique()

# Convert the unique values array to a list
unique_sectors_list = unique_sectors.tolist()

# Print the list of unique company brands
unique_sectors_list

: 

### Cleaning the Stage Column

In [ ]:
# Get unique values in the Stage column
unique_stages = data_df['Stage'].unique()

# Convert the unique values array to a list``
unique_stages_list = unique_stages.tolist()

# Print the list of unique stages
unique_stages_list

: 

In [ ]:
data_df[data_df['Stage'] == 'https://docs.google.com/spreadsheets/d/1x9ziNeaz6auNChIHnMI8U6kS7knTr3byy_YBGfQaoUA/edit#gid=1861303593']

: 

In [ ]:
# Set the values in the 'Stage' column to an empty string where the value is equal to the specified URL
data_df.loc[data_df['Stage'] == 'https://docs.google.com/spreadsheets/d/1x9ziNeaz6auNChIHnMI8U6kS7knTr3byy_YBGfQaoUA/edit#gid=1861303593', 'Stage'] = ''
data_df[data_df['Company_Brand'] == 'BuyForexOnline']

: 

In [ ]:
# Set the 'Stage' column to an empty string where the value is equal to '$6000000'
data_df.loc[data_df['Stage'] == '$6000000', 'Stage'] = ''
data_df[data_df['Company_Brand'] == 'MYRE Capital']

: 

In [ ]:
# Set 'Amount' to 60000000 for the row where 'Company_Brand' is 'MYRE Capital'
data_df.loc[data_df['Company_Brand'] == 'MYRE Capital', 'Amount'] = 60000000
data_df[data_df['Company_Brand'] == 'MYRE Capital']

: 

In [ ]:
# Select rows where the 'Stage' column contains NaN values
data_df[pd.isna(data_df['Stage'])]

: 

In [ ]:
# Convert NaN values in the 'Stage' column to an empty string ('')
data_df.loc[pd.isna(data_df['Stage']), 'Stage'] = ''

: 

In [ ]:
# Get unique values in the Stage column
unique_stages = data_df['Stage'].unique()

# Convert the unique values array to a list``
unique_stages_list = unique_stages.tolist()

# Print the list of unique stages
unique_stages_list

: 

### Cleaning HeadQuarter Column

In [ ]:
# Get unique values in the HeadQuarter column
unique_HeadQuarter = data_df['HeadQuarter'].unique()

# Convert the unique values array to a list``
unique_HeadQuarter_list = unique_HeadQuarter.tolist()

# Print the list of unique HeadQuarter
unique_HeadQuarter_list

: 

In [ ]:
# Replace 'Online Media\t#REF!' with 'Online Media'
data_df['HeadQuarter'].replace('Online Media\t#REF!', 'Online Media', inplace=True)

# Replace 'Manchester, Greater Manchester' with 'Manchester'
data_df['HeadQuarter'].replace('Manchester, Greater Manchester', 'Manchester', inplace=True)


: 

In [ ]:
# Get unique values in the HeadQuarter column
unique_HeadQuarter = data_df['HeadQuarter'].unique()

# Convert the unique values array to a list``
unique_HeadQuarter_list = unique_HeadQuarter.tolist()

# Print the list of unique HeadQuarter
unique_HeadQuarter_list

: 

In [ ]:
# Filter the data where the "HeadQuarter" column is equal to "The Nilgiris"
data_df[data_df['HeadQuarter'] == "The Nilgiris"]

: 

### Cleaning what it does column

In [ ]:
# Get unique values in the HeadQuarter column
unique_WiD = data_df['What it does'].unique()

# Convert the unique values array to a list``
unique_WiD_list = unique_WiD.tolist()

# Print the list of unique HeadQuarter
unique_WiD_list

: 

### Cleaning the Year Column

In [ ]:
# Get unique values in the Year column
unique_Year = data_df['Year'].unique()

# Convert the unique values array to a list``
unique_Year_list = unique_Year.tolist()

# Print the list of unique Year
unique_Year_list

: 

###  Cleaning the Founded Column

In [ ]:
# Get unique values in the Founded column
unique_Founded = data_df['Founded'].unique()

# Convert the unique values array to a list``
unique_Founded_list = unique_Founded.tolist()

# Print the list of unique Founded
unique_Founded_list

: 

In [ ]:
# Filter the data where the "Founded" column contains NaN values
data_df[data_df['Founded'].isna()]

: 

In [ ]:
# Calculate the average of non-NaN values in the 'Founded' column
average_founded = data_df['Founded'].mean()

# Fill NaN values in the 'Founded' column with the calculated average
data_df['Founded'].fillna(average_founded, inplace=True)

# Filter the data where the "Founded" column contains NaN values
data_df[data_df['Founded'].isna()]

: 

In [ ]:
data_df['Founded'].unique()

: 

In [ ]:
# Round all values in the 'Founded' column to the nearest whole number
data_df['Founded'] = data_df['Founded'].round()

# View unique values in the 'Founded' column after rounding
data_df['Founded'].unique()


: 

In [ ]:
# Convert all values in the 'Founded' column to integers
data_df['Founded'] = data_df['Founded'].astype(int)

# View unique values in the 'Founded' column after conversion
data_df['Founded'].unique()


: 

### Cleaning the Founders Coulmn

In [ ]:
# Get unique values in the Founders column
unique_Founders = data_df['Founders'].unique()

# Convert the unique values array to a list``
unique_Founders_list = unique_Founders.tolist()

# Print the list of unique Founded
unique_Founders_list

: 

### Cleaning the Investor Column

In [ ]:
data_df['Investor'].unique()

: 

###

### Checking to see if everything is correct

In [ ]:
data_df.info()

: 

In [ ]:
data_df.head()

: 

#### the imputation with the mean was chosen to fill the amount column

In [ ]:
# Calculate the mean of non-null values in the 'Amount' column
mean_amount = data_df['Amount'].mean()

# Round the mean to two decimal places
mean_amount_rounded = round(mean_amount, 2)

# Fill missing values in the 'Amount' column with the rounded mean
data_df['Amount'].fillna(mean_amount_rounded, inplace=True)

# Export the DataFrame to a CSV file
data_df.to_csv('./dataset/data.csv', index=False)


: 

## EXPLORATORY DATA ANALYSIS

### Loading Data with initial statistics

In [ ]:
# Load the data
data = pd.read_csv('./dataset/data.csv')

# Inspect the first few rows
data

: 

In [ ]:
data.info()

: 

### Univariate Analysis and cleaning

In [ ]:
# Check summary statistics
data.describe()

: 

In [ ]:
data['Amount'].max()

: 

In [ ]:
data_df[data_df['Amount'] == 0.0]['Amount']

: 

In [ ]:
# Calculate the mean amount excluding 0.0 values
mean_amount = data[data['Amount'] != 0.0]['Amount'].mean()

# Fill the rows where 'Amount' is 0.0 with the mean amount
data.loc[data['Amount'] == 0.0, 'Amount'] = mean_amount

: 

In [ ]:
data.info()

: 

In [ ]:
data.describe()

: 

#### Amount:
- The count indicates that there are 2716 non-null values in this column.
- The mean (average) amount is approximately $120.52 million, with a large standard deviation of $3.18 billion, suggesting a wide range of values.
- The minimum amount is $876, and the maximum amount is $150 billion.
- The 25th percentile value is $1 million, the median (50th percentile) is $5 million, and the 75th percentile is approximately $40.25 million.

#### Year:
- The count indicates that there are 2716 non-null values in this column.
- The mean year is approximately 2019.97, indicating that most data entries are from recent years.
- The standard deviation is relatively low at approximately 1.09, suggesting that the years are relatively close to the mean.
- The minimum year is 2018, and the maximum year is 2021.
- The 25th percentile year is 2020, the median (50th percentile) is 2020, and the 75th percentile year is 2021.

#### Founded:
- The count indicates that there are 2716 non-null values in this column.
- The mean founding year is approximately 2015.97, suggesting that most companies in the dataset were founded in recent years.
- The standard deviation is approximately 3.77, indicating some variability in the founding years.
- The minimum founding year is 1963, and the maximum founding year is 2021.
- The 25th percentile founding year is 2015, the median (50th percentile) is 2016, and the 75th percentile founding year is 2018.

In [ ]:
# Check data types and missing values
data.info()

: 

The dataset contains a mix of numerical and categorical data, with some missing values in several columns such as 'Sector', 'Stage', 'HeadQuarter', 'Founders', and 'Investor'. The 'Amount' column is represented as float64, 'Year' and 'Founded' columns are represented as int64, and the rest are of type object.

#### Checking for duplicates

In [ ]:
# Check for duplicates in the DataFrame
data_df[data_df.duplicated()]

: 

In [ ]:
# Count the number of duplicate rows in the DataFrame
data_df.duplicated().sum()

: 

In [ ]:
# Remove duplicate rows from the DataFrame
data_df.drop_duplicates(inplace=True)

# Count the number of duplicate rows in the DataFrame
data_df.duplicated().sum()

: 

##### Distribution of Numeric Columns:

In [ ]:
# Set up the figure and axis
fig, axes = plt.subplots(nrows=1, ncols=3, figsize=(13, 5))

# Distribution of 'Amount' column
sns.histplot(data_df['Amount'], bins=20, kde=True, ax=axes[0])
axes[0].set_title('Distribution of Amount')

# Distribution of 'Year' column
sns.histplot(data_df['Year'], bins=20, kde=True, ax=axes[1])
axes[1].set_title('Distribution of Year')

# Distribution of 'Founded' column
sns.histplot(data_df['Founded'], bins=20, kde=True, ax=axes[2])
axes[2].set_title('Distribution of Founded')

plt.tight_layout()
plt.show()


: 

##### Frequency Counts for Categorical Columns:

In [ ]:
# Frequency counts for categorical columns
print("Frequency counts for 'Sector' column:")
data_df['Sector'].value_counts()

: 

### Bivariate ANalysis

#### Correlation Analysis

##### the correlation matrix between numeric variables to identify linear relationships.

##### the correlation matrix using a heatmap for better interpretation

#### Scatter Plots

In [ ]:
# Scatter Plots
# Create scatter plots for pairs of numeric variables
sns.pairplot(data_df[['Amount', 'Year', 'Founded']])
plt.show()

: 

In [ ]:
all_data= data

: 